app/controllers/adoption_application_controller.rb

```diff
@@ -0,0 +1,47 @@
+class AdoptionApplicationController < ApplicationController
+
+   def show
+      @adoption_app = AdoptionApplication.find(params[:id])
+      
+      @pets = Pet.search(params[:search])
+
+      if params[:add_pet].present?
+         pet_id = params[:add_pet]
+         @adoption_app.add_pet_to_app(pet_id)
+      end
+
+      if params[:ownership_description].present?
```

Comment:
Since this functionality changes an attribute on the AdoptionApplication (updates the status), a more RESTful option would be to have that "Submit Application" button go to an update action in this controller. That action can then redirect back to this show page.